tsdb/chunkenc/histo.go

```diff
@@ -258,9 +258,84 @@ func (a *HistoAppender) Appendable(h histogram.SparseHistogram) ([]Interjection,
 	if !ok {
 		return nil, nil, false
 	}
+	if counterReset(a.posbuckets, h.PositiveBuckets, a.posSpans, h.PositiveSpans) {
+		return nil, nil, false
+	}
+	if counterReset(a.negbuckets, h.NegativeBuckets, a.negSpans, h.NegativeSpans) {
+		return nil, nil, false
+	}
+
 	return posInterjections, negInterjections, ok
 }
 
+// counterReset returns true if there was a counter reset for any bucket.
+// This should be called only when buckets are same or new buckets were added,
+// and does not handle the case of buckets missing.
+func counterReset(oldBuckets, newBuckets []int64, oldSpans, newSpans []histogram.Span) bool {
```

Comment:
Perhaps call it `counterResetInAnyBucket` or something to make clear it only acts on buckets.